lib/cli/github_settings/do_check.js

```diff
@@ -123,175 +135,200 @@ function do_check(subcmd, opts, args, cb) {
     );
 }
 
+function checkBranchProtection(branchName, ctx, next) {
+    assert.string(branchName, 'branchName');
+    assert.object(ctx, 'ctx');
+    assert.func(next, 'next');
+    assert.object(ctx.log, 'ctx.log');
+    assert.object(ctx.octokit, 'ctx.octokit');
+    assert.object(ctx.repo, 'ctx.repo');
+    assert.array(ctx.checkFailures, 'ctx.checkFailures');
+    assert.array(ctx.checkWarnings, 'ctx.checkWarnings');
+    assert.array(ctx.optionalBranches, 'ctx.optionalBranches');
```

Comment:
```suggestion
    assert.arrayOrString(ctx.optionalBranches, 'ctx.optionalBranches');
```

Poorman's only slightly better but still nowhere near strong-typing. :)

Reply:
Cool.